sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -72,10 +72,15 @@ case object AllTuples extends Distribution {
 /**
  * Represents data where tuples that share the same values for the `clustering`
  * [[Expression Expressions]] will be co-located in the same partition.
+ *
+ * @param requiredAllClusterKeys When true, `Partitioning` which satisfies this distribution,
+ *                               must match all `clustering` expressions in the same ordering.
  */
 case class ClusteredDistribution(
     clustering: Seq[Expression],
-    requiredNumPartitions: Option[Int] = None) extends Distribution {
+    requiredNumPartitions: Option[Int] = None,
+    requiredAllClusterKeys: Boolean = SQLConf.get.getConf(
```

Comment:
A bug here will lead us to correctness issue for streaming query, so would like to double check.

Is it 100% safe to assume this singleton takes the config for the spark session on running query? Since we have to disable this for streaming query despite the end users enable this.

```
sparkSessionForStream.conf.set(SQLConf.REQUIRE_ALL_CLUSTER_KEYS_FOR_DISTRIBUTION.key, "false")
```

Reply:
@HeartSaVioR - Actually I am less sure about it for streaming query. we are doing same thing for [co-partition check here](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala#L526). Could you help brainstorm more if there's counterexample?

Reply:
I just wanted to make sure the config value we override in IncrementalExecution is in effect here with singleton SQLConf approach. Otherwise we will encounter the problem. For co-partition ase, I guess stream-stream join is only the case for streaming (I might not be exhaustive, but at least probably exhaustive for stateful operators) and we use StatefulOpClusteredDistribution directly so it seems OK.

Reply:
Honestly I'm also confusing about the behavior of the `SQLConf.get` so that was the reason I asked it. ;) I had to fix a bug on SQLConf.get and I was totally puzzled when dealing with it.

Reply:
It gets conf from the active spark session, and in streaming we always set the correct active session AFAIK.

Reply:
Sounds great. Thanks for clarification!

Reply:
Thanks @cloud-fan for confirming.